src/python/pants/backend/python/rules/util.py

```diff
@@ -125,7 +124,15 @@ def find_packages(
     # Now find all package_data.
     for tgt, stripped_srcs in tgts_and_stripped_srcs:
         if tgt.has_field(ResourcesSources):
-            source_root = source_roots.strict_find_by_path(tgt.address.spec_path).path
+            source_roots = list(stripped_srcs.root_to_relfiles.keys())
+            if len(source_roots) != 1:
+                # A single target is expected to be under a single source root, so this
+                # should never happen, but we check to be sure.
+                raise SourceRootError(
+                    f"Expected a single source root for target "
+                    f"{tgt.address.spec} but found: {', '.join(source_roots)}. "
+                )
```

Comment:
FYI we will have failed a long time before this. The `RemovePrefix` will complain if not every file has that same prefix.

Reply:
👍 I'd still rather be defensive here.